src/blurhash/__init__.py

```diff
@@ -34,3 +41,37 @@ def encode(image_file, x_components, y_components):
         raise ValueError('Invalid x_components or y_components')
 
     return _ffi.string(result).decode()
+
+def decode(blurhash, width, height, punch = 1, mode = 'RGBA'):
+
+    if not mode in _pixel_modes:
+        raise ValueError("Invalid value for argument mode, must be either 'RGB' or 'RGBA'")
+
+    channels = _pixel_modes[mode]
+    blurhash_str = _ffi.new('char[]', bytes(blurhash, "utf-8"))
+
+    if not _lib.is_valid_blurhash(blurhash_str) :
+        raise ValueError("{} is not a valid blurhash".format(blurhash))
+
+    width_int = _ffi.cast('int', width)
+    height_int = _ffi.cast('int', height)
+    punch_int = _ffi.cast('int', punch)
+    channels_int = _ffi.cast('int', channels)
+
+    pixel_array = _ffi.new('uint8_t[]', width * height * channels)
+
+    result = _lib.create_pixels_from_blurhash(blurhash_str, width_int, height_int,
+                                    punch_int, channels, pixel_array)
+
+    if result == -1 :
+        raise Exception("Failed to decode blurhash {}".format(blurhash))
```

Comment:
I would use a subclass of `Exception` in this case, for example, `BlurhashDecodeError`. Then, it is easier to handle all possible errors with separate `expect` statements without catching other exceptions that might be unexpected.

Reply:
Yeah added `BlurhashDecodeError` sub-class to resolve this.